pallets/file-system/src/types.rs

```diff
@@ -57,11 +57,11 @@ pub struct StorageRequestMetadata<T: Config> {
     /// SPs will expect a connection request to be initiated by the user with this Peer Id.
     pub user_peer_ids: PeerIds<T>,
 
-    /// List of storage providers that can serve the data that is requested to be stored.
+    /// List of backup storage providers that can act as a data server for a storage request, serving the data that is requested to be stored.
     ///
     /// This is useful when a BSP stops serving data and automatically creates a new storage request with no user multiaddresses, since
-    /// SPs can prove and serve the data to be replicated to other BSPs without the user having this stored on their local machine.
-    pub data_server_sps: BoundedVec<ProviderIdFor<T>, MaxBspsPerStorageRequest<T>>, // TODO: Change the Maximum data servers to be the maximum SPs allowed
+    /// BSPs can prove and serve the data to be replicated to other BSPs without the user having this stored on their local machine.
+    pub data_server_sps: BoundedVec<ProviderIdFor<T>, MaxDataServers<T>>,
```

Comment:
I think this should be bounded by the `ReplicationTarget` to ensure all BSPs who confirmed storing this file can register themselves as a data server. The alternative solution would be to add an equality check for `MaxDataServers` and `ReplicationTarget` in the `integrity_test` if we wish to keep these types distinct. I would just use the former imo.